tests/mnpe_test.py

```diff
@@ -0,0 +1,282 @@
+from sbi.utils.user_input_checks_utils import MultipleIndependent
+from tests.test_utils import check_c2st
+
+
+# toy simulator for continuous data with mixed parameters
+def mixed_param_simulator(theta: Tensor) -> Tensor:
+    """Simulator for continuous data with mixed parameters.
+
+    Args:
+        theta: Parameters with mixed types - continuous and discrete.
+    Returns:
+        x: Continuous observation.
+    """
+    device = theta.device
+
+    # Extract parameters
+    a, b = theta[:, 0], theta[:, 1]
+    noise = 0.05 * torch.randn(a.shape, device=device).reshape(-1, 1)
+    return (a + 2 * b).reshape(-1, 1) + noise
+
+
+@pytest.mark.gpu
+@pytest.mark.parametrize("device", ("cpu", "gpu"))
+def test_mnpe_on_device(
+    device,
+    num_simulations: int = 100,
+):
+    """Test MNPE API on device (CPU/GPU)."""
+    num_samples = 10
+    device = process_device(device)
+
+    # Generate data with mixed parameter types
+    theta_true = torch.tensor([[0.5, 1.0]], device=device)
+    x_o = mixed_param_simulator(theta_true).to(device)
+
+    prior = MultipleIndependent(
+        [
+            BoxUniform(low=torch.zeros(1), high=torch.ones(1)),  # continuous
+            Bernoulli(probs=0.8 * torch.ones(1)),  # discrete (Bernoulli)
+        ],
+        device=device,
+    )
+    theta = prior.sample((num_simulations,))
+    x = mixed_param_simulator(theta).to(device)
+
+    trainer = MNPE(prior=prior, device=device)
+    trainer.append_simulations(theta, x).train(max_num_epochs=3)
+
+    # Test sampling on device
+    posterior = trainer.build_posterior()
+    samples = posterior.sample(
+        sample_shape=(num_samples,),
+        x=x_o,
+        show_progress_bars=False,
+    )
+    assert samples.shape == (num_samples, 2)
+
+
+def test_batched_sampling(num_simulations: int = 100):
+    """Test MNPE API with batched sampling."""
+    batch_size = 5
+    num_samples = 10
+
+    # Generate data with mixed parameter types
+    theta_true = torch.cat(
+        (
+            torch.rand(batch_size, 1),
+            torch.ones(batch_size, 1),
+        ),
+        dim=1,
+    )
+    x_o = mixed_param_simulator(theta_true)  # This will return batch_size observations
+
+    prior = MultipleIndependent(
+        [
+            BoxUniform(low=torch.zeros(1), high=torch.ones(1)),  # continuous
+            Bernoulli(probs=0.8 * torch.ones(1)),  # discrete (Bernoulli)
+        ],
+    )
+    theta = prior.sample((num_simulations,))
+    x = mixed_param_simulator(theta)
+
+    trainer = MNPE(prior=prior)
+    trainer.append_simulations(theta, x).train(max_num_epochs=1)
+
+    # Test batched sampling
+    posterior = trainer.build_posterior()
+    samples = posterior.sample_batched(
+        sample_shape=(num_samples,),
+        x=x_o,
+        show_progress_bars=False,
+    )
+    print(samples.shape)
+    assert samples.shape == (num_samples, batch_size, 2)
+
+
+@pytest.mark.parametrize("flow_model", ("mdn", "maf", "nsf", "zuko_nsf"))
```

Comment:
```suggestion
@pytest.mark.parametrize("flow_model", ("mdn", "nsf", "zuko_nsf"))
```
maf will learn a Gaussian in 1D, and it's fine if just cover mdn, nsf (nflows) and zuko.